cloudflare_utils.py

```diff
@@ -1,43 +1,42 @@
 #!/usr/bin/env python3
 
-import ipaddress
 import dns
+import ipaddress
 import sys
 
-#  TODO: fetch it from https://www.cloudflare.com/ips-v4 instead
-cloudflare_ip_ranges = [
-    "103.21.244.0/22",
-    "103.22.200.0/22",
-    "103.31.4.0/22",
-    "104.16.0.0/12",
-    "108.162.192.0/18",
-    "131.0.72.0/22",
-    "141.101.64.0/18",
-    "162.158.0.0/15",
-    "172.64.0.0/13",
-    "173.245.48.0/20",
-    "188.114.96.0/20",
-    "190.93.240.0/20",
-    "197.234.240.0/22",
-    "198.41.128.0/17"
-]
+import requests
+
+
+def get_cloudflare_ip_ranges():
+    cloudflare_ip_ranges_url = 'https://www.cloudflare.com/ips-v4'
+    try:
+        page_content = requests.get(cloudflare_ip_ranges_url, timeout=10)
+        ip_ranges_text = page_content.text
+        return [ip for ip in ip_ranges_text.split("\n") if ip]
+    except requests.exceptions.RequestException:
+        pass
```

Comment:
If we fail to retrieve IP ranges from Cloudflare, how about:

- Using the current hardcoded list
- Printing an error message on stderr `[-] Failed to retrieve Cloudflare IP ranges - using a default (possibly outdated) list`

?

Reply:
Sounds like a better idea.